sdk/aqueduct/decorator.py

```diff
@@ -473,24 +526,43 @@ def wrapped(
             )
             check_spec = CheckSpec(level=severity, function=function_spec)
 
-            new_check_artifact = wrap_spec(
+            bool_artifact = wrap_spec(
                 OperatorSpec(check=check_spec),
                 *input_artifacts,
                 op_name=name,
                 description=description,
+                execution_mode=execution_mode,
+                output_artifact_type_hint=ArtifactType.BOOL,
             )
 
-            assert isinstance(new_check_artifact, BoolArtifact)
-            new_check_artifact.set_operator_type(OperatorType.CHECK)
+            assert isinstance(bool_artifact, BoolArtifact)
+
+            bool_artifact.set_operator_type(OperatorType.CHECK)
 
-            return new_check_artifact
+            return bool_artifact
+
+        """
+        @wraps(...): updates `wrapped()` to look like `func()` by copying attributes (eg. __name__, __doc__, etc.)
+        """
+
+        @wraps(func)
+        def wrapped(
```

Comment:
we probably need to rethink the inheritance between these decorators given there are a number of similar patterns